data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourceCoordinator.java

```diff
@@ -91,11 +91,13 @@ public interface SourceCoordinator<T> {
 
     /**
      * Should be called by the source when it is shutting down to indicate that it will no longer be able to perform work on partitions,
-     * or can be called to give up ownership of its partitions in order to pick up new ones with {@link #getNextPartition(Function)} ()}.
+     * or can be called to give up ownership of a partition in order to pick up new ones with {@link #getNextPartition(Function)} ()}.
+     * This is used when source coordinator is shared by multiple threads.
+     * @param sourcePartition source partition to giveup ownership
      * @throws org.opensearch.dataprepper.model.source.coordinator.exceptions.PartitionUpdateException if the partition could not be given up due to some failure
-     * @since 2.2
+     * @since 2.8
      */
-    void giveUpPartitions();
+    void giveUpPartition(SourcePartition<T> sourcePartition);
```

Comment:
This can just take a `String partitionKey` to align with the other functions here